lib/optimizely.rb

```diff
@@ -135,8 +135,8 @@ def get_variation(experiment_key, user_id, attributes = nil)
         return nil
       end
 
-      if user_id.to_s.empty?
-        @logger.log(Logger::ERROR, 'User ID cannot be empty.')
+      if !user_id.is_a?(String) || user_id.empty?
```

Comment:
This same logic is duplicated in a few places. Can you create a validator function that we re-use?